### PR TITLE
Waterfall recipe

### PR DESCRIPTION
**Pull request by daschw**
# Description

This is the implementation of waterfall charts as a recipe as discussed with @jkrumbiegel in #2314.
Not sure, if the amount and naming of specific attributes is OK;
It adds the following examples to the docs:

```julia
using Makie

y = [6, 4, 2, -8, 3, 5, 1, -2, -3, 7]

waterfall(y)
```
example_10448004870160825484

```julia
waterfall(y, show_direction=true)
```
example_13954012298839594358

```julia
waterfall(y, show_direction=true, marker_pos=:cross, marker_neg=:hline, direction_color=:gold)
```
example_17457802745984131740

```julia
x = repeat(1:2, inner=5)
group = repeat(1:5, outer=2)
colors = Makie.wong_colors()

waterfall(x, y, dodge=group, color=colors[group])
```
example_6936188115957499170

```julia
waterfall(x, y, dodge=group, color=colors[group], show_direction=true, show_final=true)
```
example_13365502442967251696

```julia
waterfall(x, y, dodge=group, color=colors[group], show_final=true, final_color=(colors[6], 1//3))
```
example_5313543370909876466

```julia
waterfall(group, y, dodge=x, color=colors[x], stack=:x, show_direction=true)
```
example_16283516057630021035

## Type of change

Delete options that do not apply:

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Added an entry in NEWS.md (for new features and breaking changes)
- [x] Added or changed relevant sections in the documentation
- [ ] Added unit tests for new algorithms, conversion methods, etc.
- [ ] Added reference image tests for new plotting functions, recipes, visual options, etc.

### CODE REVIEW

File: src/basic_recipes/waterfall.jl [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        return (xy=xy, fillto=fillto, final=final)
+    end
+
+    fromto = lift(stack_bars, p[1], p.dodge, p.stack)
+
+    if p.show_final[]
+        final_gap = p.final_gap[] === automatic ? p.dodge[] == automatic ? 0 : p.gap : p.final_gap
+        barplot!(
+            p,
+            fromto[].final;
```
Comment (SimonDanisch) [inline src/basic_recipes/waterfall.jl:62]: You should turn all the `fromto[]`/`marker[]`/etc into: `lift(x-> x.xy, fromto)`

Comment (daschw) [reply src/basic_recipes/waterfall.jl:62]: Thanks for the hint! Does this also apply to `p.show_final[]`, `p.final_gap[]` etc?

Comment (SimonDanisch) [reply src/basic_recipes/waterfall.jl:62]: Yes, but there you should be able to just remove the []

Comment (daschw) [reply src/basic_recipes/waterfall.jl:62]: Hm, but if I remove `[]` e.g. in L58, I get
```julia
ERROR: TypeError: non-boolean (Observable{Any}) used in boolean context
```

**Comment by daschw** [comment]
bump: I added more `lift`s in the recipe as suggested by Simon. Is there anything else I should change to move this forward?

**Review by jkrumbiegel — APPROVED** [review]

**Pull request merged**